rollup/circuitcapacitychecker/libzkp/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+#![feature(once_cell)]
+
+pub mod checker {
+    use crate::utils::{c_char_to_vec, vec_to_c_char};
+    use libc::c_char;
+    use prover::zkevm::{CircuitCapacityChecker, RowUsage};
+    use serde_derive::{Deserialize, Serialize};
+    use std::cell::OnceCell;
+    use std::collections::HashMap;
+    use std::panic;
+    use std::ptr::null;
+    use types::eth::BlockTrace;
+
+    #[derive(Debug, Clone, Deserialize, Serialize)]
+    pub struct RowUsageResult {
+        pub acc_row_usage: Option<RowUsage>,
+        pub tx_row_usage: Option<RowUsage>,
+        pub error: Option<String>,
+    }
+
+    static mut CHECKERS: OnceCell<HashMap<u64, CircuitCapacityChecker>> = OnceCell::new();
+
+    /// # Safety
+    #[no_mangle]
+    pub unsafe extern "C" fn init() {
+        env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("debug"))
```

Comment:
How does this work in practice? Will Rust logs be mixed together with geth logs?

Reply:
won't be mixed together with geth logs, we can configure different levels for rust/go.

And their format will be very different.
<img width="1285" alt="image" src="https://github.com/scroll-tech/go-ethereum/assets/37070449/7c32f9a1-4fd4-4849-aabe-fcd5d35b6263">

Reply:
By "mixed together" I meant we'll see both logs on stdout, so yes. Not an issue, just wanted to clarify.